src/panoptes/pocs/camera/camera.py

```diff
@@ -636,6 +650,70 @@ def get_thumbnail(self, seconds, file_path, thumbnail_size, keep_file=False, *ar
             self.logger.warning(f'Problem getting thumbnail: {e!r}')
         return thumbnail
 
+    def wait_for_stable_temperature(self, blocking=False, *args, **kwargs):
+        """
+        Wait until camera temperature is stable for a sufficiently long period of time.
+
+        Args:
+            blocking (bool): Block until complete? Default False.
+            time_stable (Quantity): Minimum consecutive amount of time to be considered stable.
+                Default 60s.
+            sleep_delay (Quantity): Time to sleep between checks. Default 10s.
+            timeout (Quantity): Time before Timeout error is raised. Default 300s.
+        """
+        # Restart countdown if one is already in progress
+        with suppress(AttributeError):
+            if self._temperature_thread.is_alive():
+                self.logger.warning(f"Attempted to wait for stable temperature on {self}"
+                                    " while wait already in progress. Restarting countdown.")
+                self._restart_temperature_thread = True
+                self._temperature_thread.join()
```

Comment:
@wtgee I think it may still come in useful. It could save a lot of confusion in the off chance that `_check_temperature_stability` gets called before a previous call to it has returned. For instance if someone sets `cooling_enabled` then changes `target_temperature` etc.